File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int findBottomLeftValue(TreeNode root) {
        Queue<TreeNode>q = new LinkedList<>();
        q.offer(root);
        
        int res = -1;
        
        while(!q.isEmpty()){
            
            int qSize = q.size();
            
            for(int i = 0; i < qSize; i++){
                
                TreeNode front = q.poll();
                
                if(front.left != null)     q.offer(front.left);
                if(front.right != null)    q.offer(front.right);
                
                if(i == 0)  res = front.val;
                
            }
        }
        
        return res;        
    }
}

File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {        
        queue<TreeNode*>q;
        q.push(root);
        
        int res = -1;
        
        while(!q.empty()){
            
            int qSize = q.size();
            
            for(int i = 0; i < qSize; i++){
                
                TreeNode* front = q.front();
                q.pop();
                
                if(front->left)     q.push(front->left);
                if(front->right)    q.push(front->right);
                
                if(i == 0)  res = front->val;
                
            }
        }
        
        return res;
    }
};